crates/sage/src/scoring.rs

```diff
@@ -570,6 +614,15 @@ impl<'db> Scorer<'db> {
                             y_run.matched(idx);
                         }
                     }
+
+                    if self.annotate_matches {
+                        fragments_details.kinds.push(frag.kind);
+                        fragments_details.charges.push(charge as i32);
+                        fragments_details.mz_experimental.push(exp_mz);
+                        fragments_details.mz_calculated.push(calc_mz);
+                        fragments_details.fragment_ordinals.push(idx as i32);
```

Comment:
I think we should use the following code snippet here (produces b1 instead of b0, and normal y-ion numberings):
```rs
let idx = match frag.kind {
    Kind::A | Kind::B | Kind::C => idx as i32 + 1,
    Kind::X | Kind::Y | Kind::Z => {
        peptide.sequence.len().saturating_sub(1) as i32 - idx as i32
    }
};
```